src/Plugin/Create/Payload.php

```diff
@@ -94,34 +104,96 @@ public static function fromRequest(Request $request): static {
+		 *               parts: array<string>
+		 *           },
+		 *           create-def?: bool,
+		 *           options?: array<array{
+		 *               expr_type: string,
+		 *               base_expr: string,
+		 *               delim: string,
+		 *               sub_tree?: array<array{
+		 *                   expr_type: string,
+		 *                   base_expr: string
+		 *               }>
+		 *           }>
+		 *       },
+		 *       LIKE: array{
+		 *           expr_type: string,
+		 *           table?: string,
+		 *           base_expr: string,
+		 *           no_quotes: array{
+		 *               delim: bool,
+		 *               parts: array<string>
+		 *           }
+		 *       }
+		 *   }|false $payload
+		 */
 		$payload = Payload::$sqlQueryParser::parse(
 			$request->payload,
 			fn($request) => (
-				stripos(
-					$request->error,
-					"P03: syntax error, unexpected tablename, expecting \$end near 'with data'"
-				) !== false
-				&& stripos($request->payload, 'create') !== false
+				stripos($request->payload, 'create') !== false
 				&& stripos($request->payload, 'table') !== false
-				&& stripos($request->payload, 'like') !== false
-				&& stripos($request->payload, 'with') !== false
-				&& stripos($request->payload, 'data') !== false
+				&& (
+					(
+						stripos(
+							$request->error,
+							'P01: syntax error, unexpected CREATE near'
+						) !== false
+						&& stripos($request->payload, 'engine') !== false
+					) || (
+						stripos(
+							$request->error,
+							"P03: syntax error, unexpected tablename, expecting \$end near 'with data'"
+						) !== false
+						&& stripos($request->payload, 'like') !== false
+						&& stripos($request->payload, 'with') !== false
+						&& stripos($request->payload, 'data') !== false
+					)
+				)
 			), $request
 		);
 
-		if (!$payload) {
+		if (!$payload || !isset(
+			$payload['CREATE'],
+			$payload['TABLE']['no_quotes']['parts'][0],
+			$payload['TABLE']['options'][0]['base_expr']
+		)) {
 			return false;
 		}
 
-		if (isset($payload['CREATE'])
-			&& isset($payload['TABLE']['no_quotes']['parts'][0])
-			&& isset($payload['LIKE']['no_quotes']['parts'][0])
-			&& isset($payload['TABLE']['options'][0]['base_expr'])
-			&& strtolower($payload['TABLE']['options'][0]['base_expr']) === 'with data'
-		) {
-			return true;
-		}
+		static::$requestTarget = match (true) {
+			(isset($payload['LIKE']['no_quotes']['parts'][0])
+				&& strtolower($payload['TABLE']['options'][0]['base_expr']) === 'with data') => 'Handler',
+			(isset($payload['TABLE']['options'][0]['sub_tree'][0]['base_expr'])
+				&& strtolower($payload['TABLE']['options'][0]['sub_tree'][0]['base_expr']) === 'engine')
+				=> 'WithEngineHandler',
+			default => null,
+		};
 
-		return false;
+		return (static::$requestTarget !== null);
 	}
+
+	/**
+	 * @return string
+	 */
+	public function getHandlerClassName(): string {
+		return __NAMESPACE__ . '\\' . (string)static::$requestTarget;
```

Comment:
When we remove nullable, it's a good idea to remove type casting here. Using non-nullable types can simplify your code and make it more robust.